docs/glossary.md

```diff
@@ -0,0 +1,41 @@
+| Feature engineering      | Apply domain knowledge and statistical techniques to raw data to extract more information out of data and improve performance of machine learning models                                      |
+| EDA                      | Exploratory data analysis. Allows data scientist to understand dataset via cleaning, visualization, and statistical tests                                                                     |
+| ML Pipeline              | Pipeline of operations for machine learning - may include loading data, feature engineering, feature selection, model training, hyperparameter tuning, model validation, and model deployment |
+| Feature                  | Raw characteristic or numerical value of a dataset                                                                                                                                            |
+| MLOps                    | Set of practices that aims to deploy and maintain machine learning models in production reliably and efficiently. Combinination of Machine Learning and DevOps                                |
+| Dataframe                | Tabular representation of data. Often using tools such as Pandas, Spark, or Dask                                                                                                              |
+|                          |                                                                                                                                                                                               |
+| **ML Libraries / Tools** | **Definition**                                                                                                                                                                                |
+| Dask                     | Flexible library for parallel computing in Python. Often used for data engieering, data science, and machine learning                                                                         |
+| KubeFlow pipeline        | Platform for building and deploying portable, scalable machine learning (ML) workflows based on Docker containers                                                                             |
+| Sklearn                  | Open source machine learning Python library. Used for modelling, pipelines, data transformations, feature engineering, and more.                                                              |
+| Spark                    | Open source parallel processing framework for running large-scale data analytics applications across clustered computers. Often used for data engineering, data science, and machine learning |
+| XGBoost                  | Optimized distributed gradient boosting library designed to be highly efficient, flexible and portable. Implements machine learning algorithms under the Gradient Boosting framework          |
+
```

Comment:
i assume we need to add TensorFlow, Keras, PyTorch, TensorBoard (u can consult w @guy1992l  on it)